be/src/vec/columns/column_dictionary.h

```diff
@@ -264,127 +261,121 @@ class ColumnDictionary final : public COWHelper<IColumn, ColumnDictionary<T>> {
 
     ColumnPtr convert_to_predicate_column() {
         auto res = vectorized::PredicateColumnType<StringValue>::create();
-        size_t size = codes.size();
+        size_t size = _codes.size();
         res->reserve(size);
         for (size_t i = 0; i < size; ++i) {
-            auto& code = reinterpret_cast<T&>(codes[i]);
-            auto value = dict.get_value(code);
+            auto& code = reinterpret_cast<T&>(_codes[i]);
+            auto value = _dict.get_value(code);
             res->insert_data(value.ptr, value.len);
         }
-        dict.clear();
+        _dict.clear();
         return res;
     }
 
-    void convert_dict_codes() {
-        if (!is_dict_sorted()) {
-            sort_dict();
-        }
-
-        if (!is_dict_code_converted()) {
-            for (size_t i = 0; i < size(); ++i) {
-                codes[i] = dict.convert_code(codes[i]);
-            }
-            _dict_code_converted = true;
+    ColumnPtr convert_to_predicate_column_if_dictionary() override {
+        auto res = vectorized::PredicateColumnType<StringValue>::create();
```

Comment:
What's the difference between ```convert_to_predicate_column_if_dictionary``` and ```convert_to_predicate_column ```

Reply:
No difference, `convert_to_predicate_column` is useless, I'll delete it right away

Reply:
> No difference, `convert_to_predicate_column` is useless, I'll delete it right away

done